src/VisualStudio/Core/Def/Implementation/LanguageService/HACK_AbstractCreateServicesOnUiThread.cs

```diff
@@ -52,6 +52,18 @@ void IWpfTextViewConnectionListener.SubjectBuffersDisconnected(IWpfTextView text
         /// </summary>
         public static void CreateServicesOnUIThread(IComponentModel componentModel, string languageName)
```

Comment:
Hrmm... i see the concern.  If you move it, then these types may be initialized from some BG thread which might break something.  Ok.  Completely ignore my original comment.  Your chnage LGTM.